providers/implementations/rands/drbg_hmac.c

```diff
@@ -374,11 +375,12 @@ static int drbg_hmac_set_ctx_params(void *vctx, const OSSL_PARAM params[])
         return 0;
     }
 
+    prev_hmacctx = hmac->ctx;
     if (!ossl_prov_macctx_load_from_params(&hmac->ctx, params,
                                            NULL, NULL, NULL, libctx))
         return 0;
 
-    if (hmac->ctx != NULL) {
+    if (hmac->ctx != prev_hmacctx) {
```

Comment:
@paulidale the way this was written multiple calls to set_params will call this block multiple times (it gets here when params is NULL in some cases), and potentially overwrite and other params that were set via a set_ctx_params call. Not sure if this is the correct fix..

Reply:
I think it's safer as it was.  `ossl_drbg_set_ctx_params()` doesn't set anything interesting, so the code is okay.

It would be better if `ossl_drbg_set_ctx_params()` was called before this block just to be safe.  I suggest this as a better alternative.

Reply:
How does this fix the problem? i.e it is currently possible if I was to set any of strength, min_entropylen, or min_noncelen as a user via EVP_RAND_CTX_set_params(), another call to set set_params() or to EVP_RAND_instantiate() could then overwrite this value.

Reply:
I misunderstood -- however, those are all read only params for the various DRBGs.

Reply:
Which worked until I tried to make one of them writeable, so the change is probably not needed now.. But it will bite us if any become writeable